libexec/bats

```diff
@@ -26,35 +26,58 @@ help() {
   echo
 }
 
+BATS_READLINK=
+
 resolve_link() {
-  $(type -p greadlink readlink | head -1) "$1"
+  if [[ -z "$BATS_READLINK" ]]; then
+    if command -v 'greadlink' >/dev/null; then
+      BATS_READLINK='greadlink'
+    elif command -v 'readlink' >/dev/null; then
+      BATS_READLINK='readlink'
+    else
+      BATS_READLINK='true'
+    fi
+  fi
+  "$BATS_READLINK" "$1" || return 0
```

Comment:
What's the rationale for the `true` fallback? (and for returning successfully if the command fails)

Would we not want to fail if readlink is not successful? Or at least just return the original argument?

The original code would have simply failed if neither `(g)readlink` were available (or return the actual error code if the command failed: ie, if the argument were not a link). Should we not keep this behavior?

Reply:
`resolve_link` was originally called like this:

```bash
path="$(resolve_link "$name" || true)"
```

Without the `|| true` the script would exit when the target wasn't a valid symlink; the new code just pushes the "always true" result into the function itself. As before, the loop as a whole exits when `path` becomes the empty string.